src/unittest/CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+# Copyright (C) 2022 Istituto Italiano di Tecnologia (IIT)
+# All rights reserved.
+# Author: Luca Tricerri
+# This software may be modified and distributed under the terms of the
+# BSD-3-Clause license. See the accompanying LICENSE file for details.
+
+cmake_minimum_required(VERSION 3.5)
+
+include(FetchContent)
+
+#GTEST
+FetchContent_Declare(
+        googletest
+        GIT_REPOSITORY "https://github.com/google/googletest"
+        GIT_TAG        main
+)
+set(gtest_force_shared_crt ON CACHE BOOL "" FORCE)
+FetchContent_MakeAvailable(googletest)
+
+include(GoogleTest)
+
+project(unittest LANGUAGES CXX)
+
+enable_testing()
```

Comment:
This needs to be in the root `CMakeLists.txt`, not here, see https://cmake.org/cmake/help/v3.23/command/enable_testing.html .